src/imitation/algorithms/adversarial.py

```diff
@@ -411,73 +411,71 @@ def _build_disc_feed_dict(
-      discrim_kwargs: Arguments for the `DiscrimNet*` constructor.
-      init_rl_kwargs: Keyword arguments passed to `init_rl`,
-          used to initialize the RL algorithm.
-    """
-    logger.configure(folder=log_dir, format_strs=["tensorboard", "stdout"])
-    env = util.make_vec_env(
-        env_name,
-        num_vec,
-        seed=seed,
-        parallel=parallel,
-        log_dir=log_dir,
-        max_episode_steps=max_episode_steps,
-    )
-    gen_policy = util.init_rl(env, verbose=1, **init_rl_kwargs)
-
-    if use_gail:
+class GAIL(AdversarialTrainer):
+    def __init__(
+        self,
+        venv: VecEnv,
+        expert_demos: types.Transitions,
+        gen_policy: base_class.BaseRLModel,
+        *,
+        discrim_kwargs: Optional[Mapping] = None,
+        **kwargs,
+    ):
+        """Generative Adversarial Imitation Learning.
+
+        Most parameters are described in and passed to `AdversarialTrainer.__init__`.
+        Additional parameters that `GAIL` adds on top of its superclass initializer are
+        as follows:
+
+        Args:
+            discrim_kwargs: Optional keyword arguments to use while constructing the
+                DiscrimNetGAIL.
+
+        """
+        discrim_kwargs = discrim_kwargs or {}
         discrim = discrim_net.DiscrimNetGAIL(
-            env.observation_space, env.action_space, scale=scale, **discrim_kwargs
-        )
-    else:
-        rn = BasicShapedRewardNet(
-            env.observation_space, env.action_space, scale=scale, **reward_kwargs
+            venv.observation_space, venv.action_space, **discrim_kwargs
         )
-        discrim = discrim_net.DiscrimNetAIRL(
-            rn, entropy_weight=airl_entropy_weight, **discrim_kwargs
+        super().__init__(venv, gen_policy, discrim, expert_demos, **kwargs)
+
+
+class AIRL(AdversarialTrainer):
+    def __init__(
+        self,
+        venv: VecEnv,
+        expert_demos: types.Transitions,
+        gen_policy: base_class.BaseRLModel,
+        *,
+        reward_net_cls: Type[reward_net.RewardNet] = reward_net.BasicShapedRewardNet,
+        reward_net_kwargs: Optional[Mapping] = None,
+        discrim_kwargs: Optional[Mapping] = None,
+        **kwargs,
+    ):
+        """Adversarial Inverse Reinforcement Learning.
+
+        Most parameters are described in and passed to `AdversarialTrainer.__init__`.
+        Additional parameters that `AIRL` adds on top of its superclass initializer are
+        as follows:
+
+        Args:
+            reward_net_cls: Reward network constructor. The reward network is part of
+                the AIRL discriminator.
+            reward_net_kwargs: Optional keyword arguments to use while constructing
+                the reward network.
+            discrim_kwargs: Optional keyword arguments to use while constructing the
+                DiscrimNetAIRL.
+        """
+        # TODO(shwang): Maybe offer str=>Type[RewardNet] conversion like
```

Comment:
I'd weakly prefer not to do this here since it feels like it belongs in the config/UI layer, although I wouldn't object to it if others wanted it.